tests/o11y/o11y.go

```diff
@@ -31,32 +28,63 @@ var _ = framework.O11ySuiteDescribe("O11Y E2E tests", Label("o11y", "HACBS"), Pe
 		})
 		AfterAll(func() {
 			if !CurrentSpecReport().Failed() {
-				// err := f.AsKubeAdmin.CommonController.DeleteAllPipelineRunsAndTasks(testNamespace)
-				// Expect(err).NotTo(HaveOccurred())
+				err := f.AsKubeAdmin.CommonController.DeleteAllPipelineRunsAndTasks(testNamespace)
+				Expect(err).NotTo(HaveOccurred())
 				Expect(f.SandboxController.DeleteUserSignup(f.UserName)).NotTo(BeFalse())
 			}
 		})
 
-		It("E2E sample test for upload 50MB pod", Pending, func() {
-			podNameRegex := ".*-upload-50mb-pod"
-			query := fmt.Sprintf("last_over_time(container_network_transmit_bytes_total{namespace='%s', pod=~'%s'}[1h])", testNamespace, podNameRegex)
+		It("E2E test to measure Egress pod by pushing images to quay", func() {
+
+			// Get Quay Organization from ENV
+			quayOrg := utils.GetEnv("DEFAULT_QUAY_ORG", "")
+			Expect(quayOrg).ToNot(BeEmpty())
+
+			// Get Quay Token from ENV
+			quayToken := utils.GetEnv("QUAY_TOKEN", "")
+			Expect(quayToken).ToNot(BeEmpty())
+
+			_, err := f.AsKubeAdmin.CommonController.CreateRegistryAuthSecret(o11yUserSecret, testNamespace, quayToken)
+			Expect(err).ToNot(HaveOccurred())
 
+			err = f.AsKubeAdmin.CommonController.LinkSecretToServiceAccount(testNamespace, o11yUserSecret, O11ySA)
+			Expect(err).ToNot(HaveOccurred())
+
+			pipelineRun, err := f.AsKubeAdmin.O11yController.QuayImagePushPipelineRun(quayOrg, o11yUserSecret, testNamespace)
+			Expect(err).NotTo(HaveOccurred())
+
+			// Wait for the pipeline run to succeed
+			Eventually(func() bool {
```

Comment:
The inner `func` looks like something we're going to reuse often. Can you move it to the `utils` pkg?

Reply:
Done